.github/workflows/scripts/docker-setup.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Contains code to automate the Docker initialization in CI
+
+# This is mainly to be used by CI on Mac
+# After using homebrew to install docker, we need to do some magic to remove the need to interact with the GUI
+# See: https://github.com/docker/for-mac/issues/2359#issuecomment-607154849 for why we need to do things below
+set -x
+# Need to start docker if it was freshly installed or updated
+if [[ $(uname -s) = 'Darwin' ]]; then
+    if [ ! -f /Library/PrivilegedHelperTools/com.docker.vmnetd ]; then
+        echo "Making some changes to complete Docker initialization"
+        # allow the app to run without confirmation
+        xattr -d -r com.apple.quarantine /Applications/Docker.app
+
+        # preemptively do docker.app's setup to avoid any gui prompts
+        sudo-askpass /bin/cp /Applications/Docker.app/Contents/Library/LaunchServices/com.docker.vmnetd /Library/PrivilegedHelperTools/
+        sudo-askpass /bin/chmod 544 /Library/PrivilegedHelperTools/com.docker.vmnetd
+
+        # This file used to be generated as part of brew's installation
+        sudo-askpass /bin/cp .github/files/com.docker.vmnetd.plist /Library/LaunchDaemons/
```

Comment:
This used to be `sudo-askpass /bin/cp /Applications/Docker.app/Contents/Resources/com.docker.vmnetd.plist /Library/LaunchDaemons/`, however, it seems that newer Docker installations do not generate the file until the user opens the UI the first time.

Reply:
Hmm do we know why they changed it? Do we still need to do these init steps?

Reply:
I remember executing the original code (some months back) and it worked on my local machine.

I tried executing it again with the latest Docker and that file is not generated until the UI has been interacted with.

Perhaps things were different on the Mac GH runners some months ago and that file was there, however, as in my machine, I can't see it anymore.

Reply:
We still need these init steps *if* we want to keep testing `make bootstrap` which requires initializing the dev services.